python-sdk/src/astro/databases/snowflake.py

```diff
@@ -475,35 +477,68 @@ def create_table_using_native_schema_autodetection(
             },
         )
 
+    @classmethod
+    def use_quotes(cls, cols: tuple[str, ...]) -> bool:
+        """
+        With snowflake identifier we have two cases,
+
+        1. When Upper/Mixed case col names are used
+            We are required to preserver the text casing of the col names. By adding the quotes around identifier.
+        2. When lower case col names are used
+            We can use them as is
+
+        This is done to be in sync with Snowflake SQLAlchemy dialect.
+        https://docs.snowflake.com/en/user-guide/sqlalchemy.html#object-name-case-handling
+
+        Snowflake stores all case-insensitive object names in uppercase text. In contrast, SQLAlchemy considers all
+        lowercase object names to be case-insensitive. Snowflake SQLAlchemy converts the object name case during
+        schema-level communication (i.e. during table and index reflection). If you use uppercase object names,
+        SQLAlchemy assumes they are case-sensitive and encloses the names with quotes. This behavior will cause
+        mismatches against data dictionary data received from Snowflake, so unless identifier names have been truly
+        created as case sensitive using quotes (e.g. "TestDb"), all lowercase names should be used on the SQLAlchemy
+        side.
+
+        :param cols: tuple of columns
+        """
+        return any(col for col in cols if not col.islower())
```

Comment:
We should add a unit test with parameterize for this method

Reply:
Separately, a TODO for you is that you are going to create a Github issue on the double-loading for non-native path

Reply:
Added the testcases for use_quotes() function.

Reply:
@kaxil Added the ticket - https://github.com/astronomer/astro-sdk/issues/1097